packages/amazon-bedrock/src/bedrock-chat-language-model.ts

```diff
@@ -212,49 +220,50 @@ export class BedrockChatLanguageModel implements LanguageModelV1 {
           toolName: part.toolUse?.name ?? `tool-${this.config.generateId()}`,
           args: JSON.stringify(part.toolUse?.input ?? ''),
         })),
-      finishReason: mapBedrockFinishReason(response.stopReason),
+      finishReason: mapBedrockFinishReason(
+        response.stopReason as BedrockStopReason,
```

Comment:
why is the `as` needed here? shouldn't zod validation narrow the types correctly?

Reply:
The response zod schema was just parsing it as a string. I updated the zod schema so we don't need the `as` any more and we have better type info.